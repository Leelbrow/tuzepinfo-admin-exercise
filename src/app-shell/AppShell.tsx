import { Box } from "@mui/material";
import { FC } from "react";
import AppShellContent from "./components/AppShellContent";
import AppShellHeader from "./components/AppShellHeader";
import AppShellSidebar from "./components/AppShellSidebar";

const AppShell: FC = () => {
  return (
    <Box sx={{ display: "flex" }}>
      <AppShellHeader />
      <AppShellSidebar />
      <AppShellContent />
    </Box>
  );
};

export default AppShell;
