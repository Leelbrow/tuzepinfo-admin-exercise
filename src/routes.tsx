import AppShell from "./app-shell/AppShell";
import ProductListPage from "./pages/product-list/ProductListPage";

const routes = [
  {
    path: "/",
    element: <AppShell />,
    children: [
      {
        path: "products",
        element: <ProductListPage />,
      },
    ],
  },
];

export default routes;
