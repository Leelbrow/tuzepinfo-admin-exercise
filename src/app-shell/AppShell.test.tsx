import { cleanup, render, screen } from "@testing-library/react";
import { userEvent } from "@testing-library/user-event";
import { createMemoryRouter, RouterProvider } from "react-router-dom";
import { afterEach, beforeEach, describe, expect, test } from "vitest";
import routes from "../routes";
import AppShell from "./AppShell";
import menuItems from "./menu-items";

describe(AppShell.name, () => {
  let router: ReturnType<typeof createMemoryRouter>;

  beforeEach(() => {
    router = createMemoryRouter(routes);
  });

  afterEach(() => {
    cleanup();
  });

  test("renders correctly", () => {
    const { container } = render(<RouterProvider router={router} />);
    expect(container).toMatchSnapshot();
  });

  describe("navigates correctly when a menu item is clicked", () => {
    menuItems.forEach((menuItem) => {
      test(menuItem.name, async () => {
        const user = userEvent.setup();
        render(<RouterProvider router={router} />);
        await user.click(screen.getByText(menuItem.name));
        expect(router.state.location.pathname).toBe(menuItem.url);
      });
    });
  });
});
