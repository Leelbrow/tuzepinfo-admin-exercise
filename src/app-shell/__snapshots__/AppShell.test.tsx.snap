// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AppShell > renders correctly 1`] = `
<div>
  <div
    class="MuiBox-root css-k008qs"
  >
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-ivnu1y-MuiPaper-root-MuiAppBar-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
      >
        <h1
          class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-8u39c-MuiTypography-root"
        >
          Tüzépinfó admin
        </h1>
      </div>
    </header>
    <div
      class="MuiDrawer-root MuiDrawer-docked css-1f2xuhi-MuiDrawer-docked"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiDrawer-paper MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft css-12i7wg6-MuiPaper-root-MuiDrawer-paper"
      >
        <div
          class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
        />
        <div
          class="MuiBox-root css-1v8x7dw"
        >
          <ul
            class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
          >
            <li
              class="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
            >
              <div
                class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-16ac5r2-MuiButtonBase-root-MuiListItemButton-root"
                role="button"
                tabindex="0"
              >
                <div
                  class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="SellIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="m21.41 11.41-8.83-8.83c-.37-.37-.88-.58-1.41-.58H4c-1.1 0-2 .9-2 2v7.17c0 .53.21 1.04.59 1.41l8.83 8.83c.78.78 2.05.78 2.83 0l7.17-7.17c.78-.78.78-2.04-.01-2.83M6.5 8C5.67 8 5 7.33 5 6.5S5.67 5 6.5 5 8 5.67 8 6.5 7.33 8 6.5 8"
                    />
                  </svg>
                </div>
                <div
                  class="MuiListItemText-root css-tlelie-MuiListItemText-root"
                >
                  <span
                    class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                  >
                    Termékek
                  </span>
                </div>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </div>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
            >
              <div
                class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-16ac5r2-MuiButtonBase-root-MuiListItemButton-root"
                role="button"
                tabindex="0"
              >
                <div
                  class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="CategoryIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="m12 2-5.5 9h11z"
                    />
                    <circle
                      cx="17.5"
                      cy="17.5"
                      r="4.5"
                    />
                    <path
                      d="M3 13.5h8v8H3z"
                    />
                  </svg>
                </div>
                <div
                  class="MuiListItemText-root css-tlelie-MuiListItemText-root"
                >
                  <span
                    class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                  >
                    Kategóriák
                  </span>
                </div>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </div>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
            >
              <div
                class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-16ac5r2-MuiButtonBase-root-MuiListItemButton-root"
                role="button"
                tabindex="0"
              >
                <div
                  class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="StorefrontIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="m21.9 8.89-1.05-4.37c-.22-.9-1-1.52-1.91-1.52H5.05c-.9 0-1.69.63-1.9 1.52L2.1 8.89c-.24 1.02-.02 2.06.62 2.88.08.11.19.19.28.29V19c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2v-6.94c.09-.09.2-.18.28-.28.64-.82.87-1.87.62-2.89m-2.99-3.9 1.05 4.37c.1.42.01.84-.25 1.17-.14.18-.44.47-.94.47-.61 0-1.14-.49-1.21-1.14L16.98 5zM13 5h1.96l.54 4.52c.05.39-.07.78-.33 1.07-.22.26-.54.41-.95.41-.67 0-1.22-.59-1.22-1.31zM8.49 9.52 9.04 5H11v4.69c0 .72-.55 1.31-1.29 1.31-.34 0-.65-.15-.89-.41-.25-.29-.37-.68-.33-1.07m-4.45-.16L5.05 5h1.97l-.58 4.86c-.08.65-.6 1.14-1.21 1.14-.49 0-.8-.29-.93-.47-.27-.32-.36-.75-.26-1.17M5 19v-6.03c.08.01.15.03.23.03.87 0 1.66-.36 2.24-.95.6.6 1.4.95 2.31.95.87 0 1.65-.36 2.23-.93.59.57 1.39.93 2.29.93.84 0 1.64-.35 2.24-.95.58.59 1.37.95 2.24.95.08 0 .15-.02.23-.03V19z"
                    />
                  </svg>
                </div>
                <div
                  class="MuiListItemText-root css-tlelie-MuiListItemText-root"
                >
                  <span
                    class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                  >
                    Üzletek
                  </span>
                </div>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </div>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
            >
              <div
                class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-16ac5r2-MuiButtonBase-root-MuiListItemButton-root"
                role="button"
                tabindex="0"
              >
                <div
                  class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="InventoryIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M20 2H4c-1 0-2 .9-2 2v3.01c0 .72.43 1.34 1 1.69V20c0 1.1 1.1 2 2 2h14c.9 0 2-.9 2-2V8.7c.57-.35 1-.97 1-1.69V4c0-1.1-1-2-2-2m-5 12H9v-2h6zm5-7H4V4l16-.02z"
                    />
                  </svg>
                </div>
                <div
                  class="MuiListItemText-root css-tlelie-MuiListItemText-root"
                >
                  <span
                    class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                  >
                    Árukészletek
                  </span>
                </div>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </div>
            </li>
          </ul>
        </div>
      </div>
    </div>
    <main
      class="MuiBox-root css-i9gxme"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
      />
    </main>
  </div>
</div>
`;
